tests/unit/test_util.py

```diff
@@ -88,6 +93,33 @@ def test_git_commit_info(self):
         self.assertIsInstance(git_commit_info, str)
         self.assertEqual(git_commit_info[:2], "v2")
 
+    def test_have_optional_dependency(self):
+        with self.assertRaisesRegex(ModuleNotFoundError,"Optional dependency pybtex is not available. See https://docs.pybamm.org/en/latest/source/user_guide/installation/index.html#optional-dependencies for more details."):
+            pybtex = sys.modules['pybtex']
+            sys.modules['pybtex'] = None
+            pybamm.print_citations()
+        with self.assertRaisesRegex(ModuleNotFoundError,"Optional dependency tqdm is not available. See https://docs.pybamm.org/en/latest/source/user_guide/installation/index.html#optional-dependencies for more details."):
+            sys.modules['tqdm'] = None
+            model = pybamm.BaseModel()
+            v = pybamm.Variable("v")
+            model.rhs = {v: -v}
+            model.initial_conditions = {v: 1}
+            sim = pybamm.Simulation(model)
+            sim.solve([0, 1])
+        with self.assertRaisesRegex(ModuleNotFoundError,"Optional dependency anytree is not available. See https://docs.pybamm.org/en/latest/source/user_guide/installation/index.html#optional-dependencies for more details."):
```

Comment:
IIRC writing down the complete string is not required while testing, instead, you can get away with just writing a substring. In this case, just writing `"Optional dependency anytree is not available."` should work (and look much more readable).

Reply:
Done, didn't know substrings could also work. Thanks :)